ext/storage_sources/s3_store/aws_bucket_conn.cpp

```diff
@@ -0,0 +1,119 @@
+#include <aws/core/Aws.h>
+#include <aws/s3-crt/S3CrtClient.h>
+#include <aws/s3-crt/model/DeleteObjectRequest.h>
+#include <aws/s3-crt/model/ListObjectsRequest.h>
+#include <aws/s3-crt/model/Object.h>
+#include <aws/s3-crt/model/PutObjectRequest.h>
+#include "aws_bucket_conn.h"
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+/*
+ * list_buckets --
+ *     Return list of buckets from AWS account.
+ */
+bool
+aws_bucket_conn::list_buckets(std::vector<std::string> &buckets) const
+{
+    Aws::S3Crt::Model::ListBucketsOutcome outcome = m_s3_crt_client.ListBuckets();
+    if (outcome.IsSuccess()) {
+        for (Aws::S3Crt::Model::Bucket bucket : outcome.GetResult().GetBuckets())
```

Comment:
We could use `for (const auto& bucket : outcome.GetResult().GetBuckets())` instead to improve readability.